kclvm/tools/src/query/override.rs

```diff
@@ -55,16 +114,142 @@ pub fn fix_module_override(m: &mut ast::Module, o: &ast::CmdOverrideSpec) -> boo
             action: o.action.clone(),
         };
         transformer.walk_module(m);
-        transformer.has_override
+        return Ok(transformer.has_override);
+    }
+    Ok(false)
+}
+
+/// Parse override spec string to override structure.
+///
+/// parse_override_spec("alice.age=10") -> ast::OverrideSpec {
+///     pkgpath: "".to_string(),
+///     field_path: "alice.age".to_string(),
+///     field_value: "10".to_string(),
+///     action: ast::OverrideAction::CreateOrUpdate,
+/// }
+pub(crate) fn parse_override_spec(spec: &str) -> Result<ast::OverrideSpec> {
+    if spec.contains('=') {
+        // Create or update the override value.
+        let split_values = spec.splitn(2, '=').collect::<Vec<&str>>();
+        let path = split_values
+            .get(0)
+            .ok_or_else(|| invalid_spec_error(spec))?;
+        let field_value = split_values
+            .get(1)
+            .ok_or_else(|| invalid_spec_error(spec))?;
+        let (pkgpath, field_path) = split_field_path(path)?;
+        Ok(ast::OverrideSpec {
+            pkgpath,
+            field_path,
+            field_value: field_value.to_string(),
+            action: ast::OverrideAction::CreateOrUpdate,
+        })
+    } else if let Some(stripped_spec) = spec.strip_suffix('-') {
+        // Delete the override value.
+        let (pkgpath, field_path) = split_field_path(stripped_spec)?;
+        Ok(ast::OverrideSpec {
+            pkgpath,
+            field_path,
+            field_value: "".to_string(),
+            action: ast::OverrideAction::Delete,
+        })
+    } else {
+        Err(invalid_spec_error(spec))
+    }
+}
+
+/// Build a expression from string.
+fn build_expr_from_string(value: &str) -> ast::NodeRef<ast::Expr> {
```

Comment:
Consider again whether this function should be in the query crate, it feels like a part of parser or another low level crate?

Reply:
> Consider again whether this function should be in the query crate, it feels like a part of parser or another low level crate?

Have moved it into the parser crate. Besides, I have moved `MultiAssignTransformer` the into sema crate.